Public agents (#480)
elevenlabs/elevenlabs-docs: conversational-ai-sdks/conversational-ai-guide.mdx

}
```
The signed URL contains a short lived token that can be used to start the conversation, without exposing your ElevenLabs API key to your end users.
Signed Url is needed for private agents / non-public agents. 
</Tab>
<Tab title="Public Agents">
```javascript
async function getAgentId() {
    const response = await fetch('/api/getAgentId');
    const { agentId } = await response.json();
    return agentId;
}
```

> We can just provide Agent-id for public agents, no need for Signed URL

Reviewer: ```suggestion
For public agents, only the Agent-id is required, there is no need for a signed URL
```

- It provides secure, temporary access to the conversational API
- Keeps your API key secure on the backend
- Allows the frontend to connect without exposing sensitive credentials
</Tab>
<Tab title="Public Agents">
```javascript
app.get('/api/getAgentId', (req, res) => {
    const agentId = process.env.AGENT_ID;
    res.json({ 
        agentId: `${agentId}` 
    });
});
```

> We can just provide Agent-id for public agents, no need for Signed URL

Reviewer: ```suggestion
For public agents, only the Agent-id is required, there is no need for a signed URL```

# Mount static files
app.mount("/static", StaticFiles(directory="dist"), name="static")
```
    
### 3. Creating URL Endpoint
<Tabs>

>   <Tab title="SignedUrl for private agents">

Reviewer: ```suggestion
  <Tab title="Signed-Url for private agents">
```
